File: 品味Java的21種設計模式/dp_cpp/strategy/example4/LargeCustomerStrategy.cpp
#include "LargeCustomerStrategy.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace strategy
			{
				namespace example4
				{

					double LargeCustomerStrategy::calcPrice(double goodsPrice)
					{
						puts("对于大客户，统一折扣10%");
						return goodsPrice*(1-0.1);
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/strategy/example4/LargeCustomerStrategy.java
package cn.javass.dp.strategy.example4;
/**
 * 具体算法实现，为大客户计算应报的价格
 */
public class LargeCustomerStrategy implements Strategy{
	public double calcPrice(double goodsPrice) {
		System.out.println("对于大客户，统一折扣10%");
		return goodsPrice*(1-0.1);
	}
}
